apps/checker/test/scala/matchers/DictionaryMatcher.scala

```diff
@@ -0,0 +1,62 @@
+package matchers
+
+import com.gu.typerighter.model.{Category, DictionaryRule}
+import org.scalatest.flatspec.AsyncFlatSpec
+import org.scalatest.matchers.should.Matchers
+
+class DictionaryMatcherTest extends AsyncFlatSpec with Matchers {
+  val dictionaryRules = List(
+    DictionaryRule("1", "one", Category("fakeId", "fakeCategory")),
+    DictionaryRule("2", "two", Category("fakeId", "fakeCategory")),
+    DictionaryRule("3", "three", Category("fakeId", "fakeCategory"))
+  )
+  val matcher = new DictionaryMatcher(dictionaryRules)
+
+  "isTitleCase" should "match a single title case word" in {
+    val regexDoesMatch = matcher.isTitleCase("Word")
+
+    regexDoesMatch shouldBe true
+  }
+
+  "isTitleCase" should "not match a single lower case word" in {
+    val regexDoesMatch = matcher.isTitleCase("word")
+
+    regexDoesMatch shouldBe false
+  }
+
+  "isTitleCase" should "not match a single ALL CAPS word" in {
+    val regexDoesMatch = matcher.isTitleCase("WORD")
+
+    regexDoesMatch shouldBe false
+  }
+
+  "isTitleCase" should "match title case words surrounded by spaces" in {
+    val regexDoesMatch = matcher.isTitleCase("  Hey  Nice Marmot   ")
+
+    regexDoesMatch shouldBe true
+  }
+
+  "isTitleCase" should "match title case words surrounded by spaces and hyphens" in {
+    val regexDoesMatch = matcher.isTitleCase("Not-Exactly A Lightweight")
+
+    regexDoesMatch shouldBe true
+  }
+
+  "isTitleCase" should "match title case words including apostrophes" in {
+    val regexDoesMatch = matcher.isTitleCase("You're Not Wrong Walter")
+
+    regexDoesMatch shouldBe true
+  }
+
+  "isTitleCase" should "match title case words containing accents" in {
+    val regexDoesMatch = matcher.isTitleCase("Él Düderino")
+
+    regexDoesMatch shouldBe true
+  }
+
+  "isTitleCase" should "not match mixed case word sequences" in {
```

Comment:
Ooh, why not?

Reply:
It's a bit arbitrary - I just wanted it to clearly only match Title Case phrases, and e.g. a lower case word among the capitalised first-letter words would be an exception to that.

In practice, I've only seen single word dictionary matches so far so this is unlikely to manifest itself in our matches.